File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/dawms/pages/dawms_dashboard_page.py
"""
DAWMSDashboardPage: Main dashboard for DAWMS application
Provides navigation to key DAWMS workflows like Submission Intake
"""

import logging
from playwright.sync_api import Page, Locator
from dawms.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class DAWMSDashboardPage:
    """
    DAWMSDashboardPage represents the DAWMS main dashboard.
    This is the Python equivalent of the Java DAWMSDashboardPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize DAWMSDashboardPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.submission_intake_btn: Locator = page.locator(
            "button:has-text('Submission Intake')"
        )
        
        # Stability anchor: wait for dashboard to load
        pwm.wait_visible("text=DAWMS Dashboard")
        logger.info("DAWMS Dashboard Page loaded successfully.")
    
    def go_to_submission_intake(self) -> 'SubmissionIntakePage':
        """
        Navigate to Submission Intake page
        
        Returns:
            SubmissionIntakePage instance
        """
        from dawms.pages.submission_intake_page import SubmissionIntakePage
        
        self.pwm.click(self.submission_intake_btn)
        logger.info("Clicked 'Submission Intake' button.")
        return SubmissionIntakePage(self.page, self.pwm)

File: 01_projects/code/java-playwright/cbaps-dawms/basic-pom/src/test/java/com/dawms/pages/DAWMSDashboardPage.java
package com.dawms.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.dawms.library.PlaywrightManager;

/**
 * DAWMSDashboardPage: Main dashboard for DAWMS application
 * Provides navigation to key DAWMS workflows like Submission Intake
 */
public class DAWMSDashboardPage {

	private static final Logger log = LoggerFactory.getLogger(DAWMSDashboardPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator submissionIntakeBtn;

	public DAWMSDashboardPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Locator for Submission Intake button
		this.submissionIntakeBtn = page.locator("button:has-text('Submission Intake')");

		// Stability anchor: wait for dashboard to load
		pwm.waitVisible("text=DAWMS Dashboard");
		log.info("DAWMS Dashboard Page loaded successfully.");
	}

	/**
	 * Navigate to Submission Intake page
	 * @return SubmissionIntakePage
	 */
	public SubmissionIntakePage goToSubmissionIntake() {
		pwm.click(submissionIntakeBtn);
		log.info("Clicked 'Submission Intake' button.");
		return new SubmissionIntakePage(page, pwm);
	}
}
